ws/adapter.go

```diff
@@ -25,17 +26,44 @@ type Service interface {
 	mainflux.MessagePublisher
 
 	// Subscribes to channel with specified id.
-	Subscribe(uint64, Channel) error
+	Subscribe(uint64, *Channel) error
 }
 
 // Channel is used for receiving and sending messages.
 type Channel struct {
 	Messages chan mainflux.RawMessage
 	Closed   chan bool
+	closed   bool
+	mutex    sync.Mutex
+}
+
+// NewChannel instantiates empty channel.
+func NewChannel() *Channel {
+	return &Channel{
+		Messages: make(chan mainflux.RawMessage),
+		Closed:   make(chan bool),
+		closed:   false,
+		mutex:    sync.Mutex{},
+	}
+}
+
+// Send method send message over Messages channel.
+func (channel *Channel) Send(msg mainflux.RawMessage) {
+	channel.mutex.Lock()
+	defer channel.mutex.Unlock()
+
+	if !channel.closed {
+		channel.Messages <- msg
+	}
 }
 
 // Close channel and stop message transfer.
-func (channel Channel) Close() {
+func (channel *Channel) Close() {
+	channel.mutex.Lock()
+	defer channel.mutex.Unlock()
```

Comment:
Same here, but just to be consistent (performance isn't a problem here since this method is called only once), since `Send` method will be used much more.

Reply:
Same here.